assignment/js/main.js

```diff
@@ -1,3 +1,192 @@
 /* =====================
  Copy your code from Week 4 Lab 2 Part 2 part2-app-state.js in this space
 ===================== */
+
+/* LOAD MAP */
+var map = L.map('map', {
+  center: [39.9522, -75.1639],
+  zoom: 14
+});
+var Stamen_TonerLite = L.tileLayer('http://stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}.{ext}', {
+  attribution: 'Map tiles by <a href="http://stamen.com">Stamen Design</a>, <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a> &mdash; Map data &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+  subdomains: 'abcd',
+  minZoom: 0,
+  maxZoom: 20,
+  ext: 'png'
+}).addTo(map);
+
+/* MORE SETTINGS PANEL */
+var open_panel = function(clicked,panel) {
+  panel.css('display','block');
+  clicked.text('less settings');
+}
+$('.more-button').click(function(){
+  let panel = $('#more');
+  if (panel.css('display') == "none") {
+    open_panel($(this),panel);
+  } else {
+    panel.css('display','none');
+    $(this).text('more settings');
+  }
+});
+$('#auto').change(function() {
+  if ($('#auto').prop("checked")) {
+    $('.auto').prop('selected','true');
+  }
+});
+$('.coord').click(function() {
+  let xy = [];
+  $('.coord option:selected').each(function() {
+    xy.push($(this).val() === "");
+  });
+  if (xy.includes(false)) {
+    $('#auto').prop('checked',false);
+  }
+});
+
+/* GET URL INPUT */
+
+function isURL(str) {
+  var pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
+  '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+[a-z]{2,}|'+ // domain name
+  '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
+  '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
+  '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
+  '(\\#[-a-z\\d_]*)?$','i'); // fragment locator
+  return pattern.test(str);
+}
+
+var startJSON = function() {
+  let obj = {};
+  obj['url'] = $('#JSON').val();
+  if (isURL(obj['url'])) {
+    $.ajax({
+      url     : obj['url'],
+      dataType: 'json',
+      success : function(data,textStatus,jqXHR) {
+        window.success = true;
+        window.data = data; // not kosher, but too lazy to wrap everything out of reach
```

Comment:
Written directly to window, giving access to this variable through the console. Would be better to define a variable out of the global scope.

Reply:
Actually, just realized that all variables defined are accessible through the console. The whole thing should be wrapped in `$(document).ready()`.